hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/DecodingFacade.java

```diff
@@ -1,33 +1,255 @@
+					final var nftTransferBuilder = NftTransfer.newBuilder();
+					nftTransferBuilder.setSenderAccountID(convertAddressBytesToAccountID((byte[]) nftTransferDecoded.get(0)));
+					nftTransferBuilder.setReceiverAccountID(convertAddressBytesToAccountID((byte[]) nftTransferDecoded.get(1)));
+					nftTransferBuilder.setSerialNumber((long) nftTransferDecoded.get(2));
+					tokenTransferBuilder.addNftTransfers(nftTransferBuilder.build());
+				}
+
+				builder.addTokenTransfers(tokenTransferBuilder.build());
+			}
+		}
+	}
+
+	public static void decodeTransferTokens(final Bytes input) {
+		final Tuple decodedArguments =
+				TRANSFER_TOKENS_FUNCTION.getOutputs().decode(input.slice(FUNCTION_SELECTOR_BYTES_LENGTH,
+						input.size() - FUNCTION_SELECTOR_BYTES_LENGTH).toArray());
+
+		final var tokenID = convertAddressBytesToTokenID((byte[]) decodedArguments.get(0));
+		final var accountIDs = decodeAccountIDsFromBytesArray((byte[][]) decodedArguments.get(1));
+		final var amountsConverted = convertLongArrayToBigIntegerArray((long[]) decodedArguments.get(2));
+	}
+
+	public static void decodeTransferToken(final Bytes input) {
+		final Tuple decodedArguments =
+				TRANSFER_TOKEN_FUNCTION.getOutputs().decode(input.slice(FUNCTION_SELECTOR_BYTES_LENGTH,
+						input.size() - FUNCTION_SELECTOR_BYTES_LENGTH).toArray());
+
+		final var tokenID = convertAddressBytesToTokenID((byte[]) decodedArguments.get(0));
+		final var fromAccountId = convertAddressBytesToAccountID((byte[]) decodedArguments.get(1));
+		final var toAccountId = convertAddressBytesToAccountID((byte[]) decodedArguments.get(2));
+		final var amountConverted = BigInteger.valueOf((long) decodedArguments.get(3));
+	}
+
+	public static void decodeTransferNFTs(final Bytes input) {
+		final Tuple decodedArguments =
+				TRANSFER_NFTS_FUNCTION.getOutputs().decode(input.slice(FUNCTION_SELECTOR_BYTES_LENGTH,
+						input.size() - FUNCTION_SELECTOR_BYTES_LENGTH).toArray());
+
+		final var tokenID = convertAddressBytesToTokenID((byte[]) decodedArguments.get(0));
+		final var senderIDs = decodeAccountIDsFromBytesArray((byte[][]) decodedArguments.get(1));
+		final var receiverIDs = decodeAccountIDsFromBytesArray((byte[][]) decodedArguments.get(2));
+		final var serialNumbers = convertLongArrayToBigIntegerArray((long[]) decodedArguments.get(3));
+	}
+
+	public static void decodeTransferNFT(final Bytes input) {
+		final Tuple decodedArguments =
+				TRANSFER_NFTS_FUNCTION.getOutputs().decode(input.slice(FUNCTION_SELECTOR_BYTES_LENGTH,
+						input.size() - FUNCTION_SELECTOR_BYTES_LENGTH).toArray());
+
+		final var tokenID = convertAddressBytesToTokenID((byte[]) decodedArguments.get(0));
+		final var senderID = convertAddressBytesToAccountID((byte[]) decodedArguments.get(1));
+		final var recipientID = convertAddressBytesToAccountID((byte[]) decodedArguments.get(2));
+		final var serialNumber = BigInteger.valueOf((long) decodedArguments.get(3));
+	}
+
+	public static void decodeMintToken(final Bytes input) {
```

Comment:
If we change the `decodeMint` method to return a generic `MintInput` then we don't need this method and we can remove it.